hbase-server/src/main/java/org/apache/hadoop/hbase/wal/OutputSink.java

```diff
@@ -91,6 +91,15 @@ public synchronized void startWriterThreads() {
     }
   }
 
+  public synchronized void restartWriterThreadsIfNeeded() {
+    for (WriterThread t : writerThreads) {
+      if (!t.isAlive()){
+        LOG.debug("restarting thread" + t);
+        t.start();
```

Comment:
I didn't think you could restart a dead thread. You have to create a new one?

Reply:
You are absolutely. Thanks for pointing it out.